### PR TITLE
CVE-2017-7659

### PR DESCRIPTION
**Pull request by sfm2686**

### CODE REVIEW

File: cves/CVE-2017-7659.yml [hunk 1]
```diff
@@ -47,7 +47,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  HTTP/2 requests could be crafted in a way that could cause mod_http2 to
```
Comment (rmnill202) [inline cves/CVE-2017-7659.yml]: Maybe clarify what mod_http2 is (ie: package, file, module, etc...)

File: cves/CVE-2017-7659.yml [hunk 2]
```diff
@@ -220,4 +232,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
```
Comment (rmnill202) [inline cves/CVE-2017-7659.yml:244]: Unable to leave a comment directly on any of the `lessons` fields directly, so I'll just place it here.

Perhaps `defense in depth` applies? Could probably argue that if we're using a separate, complex module for performing some action, maybe we need another layer of validation around it to ensure that it doesn't crash/return garbage data?

**Comment by rmnill202** [comment]
+2 upvotes

File: cves/CVE-2017-7659.yml (hunk 1 above)
Comment (sfm2686) [reply cves/CVE-2017-7659.yml]: That might be helpful, good point!

**Comment by sfm2686** [comment]
Thanks for the upvote! But can one person add more than one upvote for the same CVE? @rmnill202

File: cves/CVE-2017-7659.yml (hunk 2 above)
Comment (sfm2686) [reply cves/CVE-2017-7659.yml:244]: I think that's a good point, I was on the fence about arguing for defense in depth for this specific CVE but im glad someone else thinks so too.

File: cves/CVE-2017-7659.yml [hunk 3]
```diff
@@ -115,11 +121,13 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer:
-  date:
-  automated:
-  google:
-  contest:
+  answer: |
+    Someone named Robert Swiecki reported the issue to the security team. The
```
Comment (s-newman) [inline cves/CVE-2017-7659.yml:130]: Did the security team discuss how it was found? Also, did Robert Swiecki write a blog post about his vulnerability-hunting process somewhere? I found a blog post written by the credited reporter for one of my vulnerabilities that helped shed light on the reporter's fuzzing process that he used to find the CVE.

File: cves/CVE-2017-7659.yml [hunk 4]
```diff
@@ -139,8 +147,11 @@ interesting_commits:
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-    - commit:
-      note:
+    - commit: 29f50cf567423ca354ee0d5fa0dfb015c555cf55
+      note: |
```
Comment (s-newman) [inline cves/CVE-2017-7659.yml:156]: I agree with you, that does seem interesting. Was there any rationale in the commit message or the comments for why they replaced the module?

**Comment by rmnill202** [comment]
@sfm2686 I believe we each provide upvotes to the vulnerabilities that we are assigned to review based on how interesting they are!

File: cves/CVE-2017-7659.yml [hunk 5]
```diff
@@ -47,7 +47,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  HTTP/2 requests could be crafted in a way that could cause mod_http2 to
+  dereference a NULL pointer and crash the server process.
```
Comment (s-newman) [inline cves/CVE-2017-7659.yml]: Some more information about how the HTTP/2 request could cause the server to dereference a null pointer could be helpful. How does the `mod_http2` module parse the request incorrectly?

**Comment by s-newman** [comment]
+1 upvote.

File: cves/CVE-2017-7659.yml (hunk 3 above)
Comment (sfm2686) [reply cves/CVE-2017-7659.yml]: I would have added that had I found it during my research.

File: cves/CVE-2017-7659.yml (hunk 5 above)
Comment (sfm2686) [reply cves/CVE-2017-7659.yml]: Will be included in my next push.

File: cves/CVE-2017-7659.yml (hunk 3 above)
Comment (sfm2686) [reply cves/CVE-2017-7659.yml:130]: While Robert does have a blog, I did not find any entries for him about this specific issue.

File: cves/CVE-2017-7659.yml (hunk 4 above)
Comment (sfm2686) [reply cves/CVE-2017-7659.yml:156]: No, the commit message was _**mod_http2/mod_proxy_http2: no longer reusing h2_request.o for proxy module**_ of the VCC commit.

**Pull request merged**